src/EditorFeatures/Core/Implementation/IntelliSense/QuickInfo/IntellisenseQuickInfoBuilder.cs

```diff
@@ -79,16 +107,67 @@ internal static class IntellisenseQuickInfoBuilder
             }
 
             var content = new ContainerElement(
-                                ContainerElementStyle.Stacked,
+                                ContainerElementStyle.Stacked | ContainerElementStyle.VerticalPadding,
                                 elements);
 
             return new IntellisenseQuickInfoItem(trackingSpan, content);
         }
 
-        private static ClassifiedTextElement BuildClassifiedTextElement(QuickInfoSection section)
+        private static IEnumerable<object> BuildClassifiedTextElements(QuickInfoSection section)
```

Comment:
so, this actually doesnt consume any section info except taggedParts. which is good. can this be changed to extension method so that other can reuse? like ToClassifiedTexts(this IEnumerable<TaggedText>) ?

Reply:
The return type contains both `ClassifiedTextElement` and `ContainerElement`, which do not share a base type other than `object`. This make the return value of `BuildClassifiedTextElements` particularly unpleasant to use. The current form allows me to encapsulate the ugliness so the untyped enumeration is not visible outside the containing `IntellisenseQuickInfoBuilder` type. I'm not opposed to refactoring this code, but I don't want to expose it with a weakly-typed signatures. Thoughts on how to resolve it?

Reply:
is ClassifiedTextElement thing only for quick info? or this behavior (format) we only want to use in quick info? otherwise, sharing implementation seems good to me.

Reply:
These are only for Quick Info.

Reply:
I looked into refactoring this, specifically by wrapping raw `ClassifiedTextElement` paragraphs into a single-element `ContainerElement`. While these would look the same in Quick Info, they would end up adding new elements to the WPF rendering tree. Since the code is not used outside this type and the documentation should keep the current approach clear, it doesn't seem like extracting an extension method for this would help clarity at this time.